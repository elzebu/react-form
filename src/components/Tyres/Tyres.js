import React, {useEffect, useState, useRef} from 'react';
import PropTypes from 'prop-types'

import Tyre from '../Tyre/Tyre'

function Tyres({ brands, tyres }) {

    const [filter, setFilter] = useState('')
    const inputRef = useRef()
    
    // Set focus on search on first load
    useEffect(() => inputRef.current.focus(), [])

    const tyresDOM = tyres
        .filter(tyre => tyre.name.toLowerCase().includes(filter.toLowerCase()))
        .map(tyre => {
            const brand = brands.find(brand => brand.id === tyre.brandId)
            tyre.brand = brand ? brand : null
            return tyre
        })
        .map(tyreWithBrand => {
            return <Tyre data={tyreWithBrand} key={tyreWithBrand.id} />
        })

    return (
        <>
            <h2>Liste des pneus</h2>
            <input type="text" ref={inputRef} className="w-100" placeholder="rechercher" onChange={(evt) => setFilter(evt.target.value)} />
            {tyresDOM}
        </>
    )
}

Tyres.propTypes = {
    brands: PropTypes.array.isRequired,
    tyres: PropTypes.array.isRequired
}

export default Tyres