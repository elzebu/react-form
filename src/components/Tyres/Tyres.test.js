import React from 'react'
import { create } from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react'

import Tyres from './Tyres'


describe('<Tyres />', () => {


    it('matches previous snapshot', () => {

        const component = create(
            <Tyres tyres={[]} brands={[]} />,
        )
        let brand = component.toJSON()
        expect(brand).toMatchSnapshot()
    })

    it('display all tyres on init', () => {
        const tyres = [{
            id: 1,
            name: 'tyre 1',
            description: 'description',
            brandId: 1
        }, {
            id: 2,
            name: 'tyre 1',
            description: 'description',
            brandId: 1
        }]

        const brands = [{
            id: 1,
            name: 'test brand',
            src: 'src'
        }]

        const { container } = render(<Tyres tyres={tyres} brands={brands} />)

        const tyreList = container.querySelectorAll('.tyre')

        expect(tyreList.length).toBe(tyres.length)
    })

    it('filter list on input change', () => {
        // Arrange
        const tyres = [{
            id: 1,
            name: 'tyre 1',
            description: 'description',
            brandId: 1
        }, {
            id: 2,
            name: 'tyre 2',
            description: 'description',
            brandId: 1
        }]
        const brands = [{
            id: 1,
            name: 'test brand',
            src: 'src'
        }]
        const { container } = render(<Tyres tyres={tyres} brands={brands} />)
        const input = container.querySelector('input')
       
        // act
        fireEvent.change(input, { target: { value: 'tyre 1' } })

        // assert
        const tyreList = container.querySelectorAll('.tyre')
        expect(tyreList.length).toBe(1)
    })

})

