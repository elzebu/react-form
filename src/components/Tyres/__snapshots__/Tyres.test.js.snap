// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tyres /> matches previous snapshot 1`] = `
Array [
  <h2>
    Liste des pneus
  </h2>,
  <input
    className="w-100"
    onChange={[Function]}
    placeholder="rechercher"
    type="text"
  />,
]
`;
